src/Plugin/Sampler/MediaSource.php

```diff
@@ -80,19 +92,47 @@ public function collect() {
 
     foreach ($types as $name => $type) {
       if ($type->getEntityType()->getProvider() === 'media_entity') {
+        $source_configuration['source_field'] = $this->getSourceFieldIndex($type->getTypeConfiguration()['source_field'], $name);
         $source = $type->getType()->getPluginId();
       }
       else {
+        $source_configuration['source_field'] = $this->getSourceFieldIndex($type->get('source_configuration')['source_field'], $name);
         $source = $type->getSource()->getPluginId();
       }
 
       $mapping = $this->groupMapping->getGroupMapping($entityTypeId, $name);
       $this->collectedData[$mapping]['source'] = $source;
+      $this->collectedData[$mapping]['source_configuration'] = $source_configuration;
     }
 
     return $this->collectedData;
   }
 
+  /**
+   * Retrieve the index of the source field.
+   *
+   * This implementation mimics the generation of fields in Bundle.php.
+   *
+   * @param string $sourceField
+   *   The source field name.
+   * @param string $bundle
+   *   The media bundle.
+   *
+   * @return int
+   *   The index of the source field
+   */
+  protected function getSourceFieldIndex(string $sourceField, string $bundle) {
+    $entityTypeId = $this->entityTypeId();
+    $baseFields = $this->entityFieldManager->getBaseFieldDefinitions($entityTypeId);
+
+    $fields = array_diff_key(
```

Comment:
I think diff is not necessary here? Since we are doing `array_search` on keys after, this diff will just make a search on a smaller array, but not that it will remove something that could lead to wrong results. Right?

Reply:
No, we have to remove base fields, because they will also be removed, when the fields are added. Otherwise, array_search would return a different index. See bundle.php line 140 for comparison.